cranelift/codegen/meta/src/gen_legalizer.rs

```diff
@@ -696,6 +696,38 @@ fn gen_isa(
     fmtln!(fmt, "];");
 }
 
+fn generate_shared(
+    fmt: &mut Formatter,
+    transform_groups: &TransformGroups,
+    shared_group_names: HashSet<&'static str>,
+) {
+    // Generate shared legalize groups.
+    let mut type_sets = UniqueTable::new();
+    let mut sorted_shared_group_names = Vec::from_iter(shared_group_names);
+    sorted_shared_group_names.sort();
+    for group_name in &sorted_shared_group_names {
+        let group = transform_groups.by_name(group_name);
+        gen_transform_group(group, transform_groups, &mut type_sets, fmt);
+    }
+    gen_typesets_table(&type_sets, fmt);
+}
+
+pub(crate) fn generate_new_backend(
+    transform_groups: &TransformGroups,
+    filename: &str,
+    out_dir: &str,
+) -> Result<(), error::Error> {
+    let mut shared_group_names = HashSet::new();
+    shared_group_names.insert("expand");
+
+    // Generate shared legalize groups.
+    let mut fmt = Formatter::new();
+    generate_shared(&mut fmt, transform_groups, shared_group_names);
+    fmt.update_file(filename, out_dir)?;
+
+    Ok(())
+}
```

Comment:
Not necessarily in this PR (but you could if you wanted to): what do you think of using a trait to describe that things like these transform groups can be generated? Then the awkward `(isas, new_backend_isas)` could be done away with in `build.rs`. I think that type of thing makes the code easier to use, even if eventually we get rid of the old/new backend distinction.

Reply:
(You still need an enum like the following but that still seems better than the extra logic above:)
```
enum BackendType {
  New(TransformGroups),
  Old(TransfomGroups),
}
impl Generator for BackendType ...
```